### PR TITLE
Fixed problem in filter with locations that have one child

### PR DESCRIPTION
**Pull request by kkrampa**

### CODE REVIEW

File: corehq/apps/locations/static/locations/ko/location_drilldown.async.js [hunk 1]
```diff
@@ -143,7 +143,11 @@ function LocationModel(data, root, depth, func, withAllOption) {
 
   // helpers to account for the 'all' meta-entry
   this.num_children = ko.computed(function() {
-      return (this.children().length === 0 ? 0 : this.children().length - 1);
```
Comment (kkrampa) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: Subtracting -1 excludes all option. It should be done only if there is all option.

**Comment by kkrampa** [comment]
@snopoke @czue

File: corehq/apps/locations/static/locations/ko/location_drilldown.async.js [hunk 2]
```diff
@@ -143,7 +143,11 @@ function LocationModel(data, root, depth, func, withAllOption) {
...
+      var length = this.children().length;
+      if (withAllOption) {
+          length -= 1;
+      }
+      return (length === 0 ? 0 : length);
```
Comment (snopoke) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: What's the purpose of this? It doesn't seem to do anything different from `return length`

Comment (kkrampa) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: It was in original code but I agree it doesn't make sense.

**Comment by kkrampa** [comment]
@snopoke Updated

File: corehq/apps/locations/static/locations/ko/location_drilldown.async.js [hunk 3]
```diff
@@ -143,7 +143,11 @@ function LocationModel(data, root, depth, func, withAllOption) {
 
   // helpers to account for the 'all' meta-entry
   this.num_children = ko.computed(function() {
-      return (this.children().length === 0 ? 0 : this.children().length - 1);
+      var length = this.children().length;
+      if (withAllOption) {
+          length -= 1;
```
Comment (gcapalbo) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: should this still be subtracting 1 if length is 0? it looks like the previous code always returned a non-negative number

Comment (kkrampa) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: It seems to me that with 'all' entry it should be always greater than 0 but I don't know original purpose of this so I will revert my last commit.

Comment (kkrampa) [inline corehq/apps/locations/static/locations/ko/location_drilldown.async.js]: @gcapalbo Please, Ignore my comment above. I have just realised what you mean.

**Comment by kkrampa** [comment]
@gcapalbo @snopoke Updated.

**Comment by snopoke** [comment]
:+1: once tests pass

**Pull request merged**